dask/array/top.py

```diff
@@ -0,0 +1,724 @@
+import collections
+import itertools
+import numbers
+
+import numpy as np
+import toolz
+
+from .. import base, core, sharedict, utils
+from ..compatibility import apply
+from ..delayed import to_task_dask
+from ..optimization import SubgraphCallable
+
+
+def subs(task, substitution):
```

Comment:
@jcrist  now that this has been moved within a more focused module do you still want it to have a different name?

Reply:
No, that's fine. Thanks.